packages/api/pages/api/languages/[code]/verses/[verseId]/next-unapproved-verse.ts

```diff
@@ -0,0 +1,40 @@
+import { GetNextUnapprovedVerse } from '@translation/api-types';
+import createRoute from '../../../../../../shared/Route';
+import { client } from '../../../../../../shared/db';
+import { authorize } from '../../../../../../shared/access-control/authorize';
+
+type NextUnapprovedVerseResult = [{ nextUnapprovedVerseId: string }];
+
+export default createRoute<{ code: string; verseId: string }>()
+  .get<void, GetNextUnapprovedVerse>({
+    authorize: authorize((req) => ({
+      action: 'translate',
+      subject: 'Language',
+      subjectId: req.query.code,
+    })),
+    async handler(req, res) {
+      const language = await client.language.findUnique({
+        where: {
+          code: req.query.code,
+        },
+      });
+      if (!language) {
+        return res.notFound();
+      }
+
+      const [result] = await client.$queryRaw<NextUnapprovedVerseResult>`
+        WITH language_gloss as 
+          (SELECT * FROM "Gloss" 
+            WHERE "Gloss"."languageId" = ${language.id}::uuid)
+        SELECT "Verse"."id" as "nextUnapprovedVerseId" FROM "Verse"
```

Comment:
Since we are just using the id of the verse, we don't need to query and join the Verse table. Just filter and return using "Word"."verseId" instead of "Verse".id. One less join is always a good thing

Reply:
Done.

Reply:
Done.